File: sdk/java/io.stockx.sdk.examples/src/main/java/io/stockx/sdk/examples/endpoints/ProductLookupExample.java
package io.stockx.sdk.examples.endpoints;

import java.util.List;

import io.stockx.client.ApiClient;
import io.stockx.client.ApiException;
import io.stockx.client.ApiResponse;
import io.stockx.client.api.StockXApi;
import io.stockx.client.model.LoginRequest;
import io.stockx.client.model.LoginResponse;
import io.stockx.client.model.ProductInfo;
import io.stockx.sdk.examples.base.Environment;
import io.stockx.sdk.examples.base.ExampleConstants;
import io.stockx.sdk.examples.base.StockXClient;

/**
 * <p>
 * Demonstrates looking up a specific product on the market.
 * </p>
 */
public class ProductLookupExample {

    public static void main(String[] args) {

        ApiClient apiClient = StockXClient.create(ExampleConstants.AWS_API_KEY, Environment.STAGING);

        StockXApi stockx = new StockXApi(apiClient);

        LoginRequest loginReq = new LoginRequest()
                        .email(ExampleConstants.STOCKX_USERNAME)
                       .password(ExampleConstants.STOCKX_PASSWORD);

        try {
            ApiResponse<LoginResponse> loginWithHttpInfo = stockx.loginWithHttpInfo(loginReq);
            List<String> list = loginWithHttpInfo.getHeaders().get(ExampleConstants.JWT_HEADER);

            String authToken = list.get(0);
            apiClient.addDefaultHeader("jwt-authorization", authToken);

            ProductInfo productInfo = stockx.lookupProduct("nike", "7");
            System.out.println("Lookup result: " + productInfo);
        } catch (ApiException e) {
            e.printStackTrace();
        }
    }

}

File: sdk/csharp/src/IO.StockX.Examples/ProductLookupExample.cs
using System;
using System.Diagnostics;
using IO.StockX.Api;
using IO.StockX.Client;
using IO.StockX.Model;

namespace Example
{
    /**
    * <p>Demonstrates looking up a specific product on the market.</p>
    */
    public class GetProductInfoExample
    {
        static public void Main()
        {
            // Configure API key authorization: api_key
            Configuration.Default.AddApiKey("x-api-key", ExampleConstants.AWS_API_KEY);

            var stockx = new StockXApi();
                stockx.Configuration.BasePath = ExampleConstants.STOCKX_ENDPOINT;
            
            var login = new LoginRequest(ExampleConstants.STOCKX_USERNAME, ExampleConstants.STOCKX_PASSWORD);
            
            try
            {
                // Login and fetch the jwt header for authentication use in the request
                ApiResponse<LoginResponse> result = stockx.LoginWithHttpInfo(login);
                var jwt = result.Headers[ExampleConstants.JWT_HEADER];

                stockx.Configuration.DefaultHeader["jwt-authorization"] = jwt;

                var productInfo = stockx.LookupProduct("nike", "7");

                Console.WriteLine(productInfo);
            }
            catch (Exception e)
            {
                Console.WriteLine(e.ToString());
                Console.WriteLine("Exception when calling StockXApi.LookupProduct: " + e.Message );
            }
        }
    }
}